routes/web/project/billing.rb

```diff
@@ -69,9 +69,35 @@ class CloverWeb
         r.redirect @project.path + "/billing"
       end
 
+      # Pre-authorize card to check if it is valid, if so
+      # authorization won't be captured and will be refunded immediately
+      begin
+        customer_stripe_id = setup_intent["customer"]
+        payment_intent = Stripe::PaymentIntent.create({
+          amount: 100, # 100 cents to charge $1.00
+          currency: "usd",
+          confirm: true,
+          off_session: true,
+          capture_method: "manual",
+          customer: customer_stripe_id,
+          payment_method: stripe_id
+        })
+
+        if payment_intent.status != "requires_capture"
+          raise "Authorization failed"
+        end
+
+        Stripe::PaymentIntent.cancel(payment_intent.id)
+      rescue
+        # Log and redirect if Stripe card error or our manual raise
+        Clog.emit("Couldn't pre-authorize card") { {card_authorization: {project_id: @project.id, customer_stripe_id: customer_stripe_id}} }
+        flash["error"] = "Couldn't pre-authorize 1$ from your card. Please try again or contact support."
```

Comment:
> In most English-speaking countries that use that symbol, it is placed to the left of the amount specified, e.g. "$1", read as "one dollar".
https://en.wikipedia.org/wiki/Dollar_sign

```suggestion
        flash["error"] = "Couldn't pre-authorize $1 from your card. Please try again or contact support."
```